opensrp-reveal/src/main/java/org/smartregister/reveal/view/EditFociBoundaryActivity.java

```diff
@@ -0,0 +1,328 @@
+
+        drawingManager.addOnDrawingCircleClickListener(new OnDrawingCircleClickListener() {
+            @Override
+            public void onCircleClick(@NonNull Circle circle) {
+                Toast.makeText(EditFociBoundaryActivity.this,
+                        getString(R.string.circle_clicked), Toast.LENGTH_LONG).show();
+                deleteBtn.setEnabled(drawingManager.getCurrentKujakuCircle() != null);
+                presenter.onEditPoint();
+            }
+
+            @Override
+            public void onCircleNotClick(@NonNull LatLng latLng) {
+                Toast.makeText(EditFociBoundaryActivity.this,
+                        getString(R.string.circle_not_clicked), Toast.LENGTH_LONG).show();
+                deleteBtn.setEnabled(false);
+            }
+        });
+
+        drawingManager.addOnKujakuLayerLongClickListener(new OnKujakuLayerLongClickListener() {
+            @Override
+            public void onKujakuLayerLongClick(@NonNull KujakuLayer kujakuLayer) {
+
+                if (drawingManager.isDrawingEnabled()) {
+                    savePointBtn.setText(R.string.save_point);
+                }
+            }
+        });
+    }
+
+    protected void setUpToolbar() {
+        toolbar = this.findViewById(R.id.edit_boundary_toolbar);
+        toolbar.setTitle(R.string.edit_boundary);
+        this.setSupportActionBar(toolbar);
+        this.getSupportActionBar().setDisplayHomeAsUpEnabled(false);
+    }
+
+    private void enabledrawingMode(MapboxMap mapboxMap) {
+        boundaryLayer.disableLayerOnMap(mapboxMap);
+        if (drawingManager != null) {
+            if (!drawingManager.isDrawingEnabled()) {
+                if (drawingManager.startDrawing( boundaryLayer)) {
+                    savePointBtn.setText(R.string.save_point);
+                }
+            } else {
+                drawingManager.stopDrawingAndDisplayLayer();
+            }
+        } else {
+            Log.e(TAG, "Drawing manager instance is null");
+        }
+
+        deleteBtn.setEnabled(false);
+    }
+
+    @Override
+    public void toggleButtons(EditBoundaryState state) {
+        switch (state) {
+            case EDITTING:
+                cancelBtn.setVisibility(View.GONE);
+                saveBoundaryBtn.setVisibility(View.GONE);
+                deleteBtn.setVisibility(View.VISIBLE);
+                savePointBtn.setVisibility(View.VISIBLE);
+                break;
+            case START:
+            case FINISHED:
+            default:
+                cancelBtn.setVisibility(View.VISIBLE);
+                saveBoundaryBtn.setVisibility(View.VISIBLE);
+                deleteBtn.setVisibility(View.GONE);
+                savePointBtn.setVisibility(View.GONE);
+                break;
+        }
+    }
+
+    @Override
+    public void exitEditBoundaryActivity() {
+        if (drawingManager != null) {
+            if (drawingManager.isDrawingEnabled()) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5681838)